Feature/inventory

sharon504 opened the pull request:
Related Issue(s): #49
- **What (has changed):**
  - Added new `Items` module with CRUD operations for managing inventory items (HSN code, name, description, unit, category)
  - Added new `Inventory` module with CRUD operations for tracking item quantities per party
  - Added `Party` and `Company` entities to support multi-party inventory tracking
  - Integrated response metadata decorator for paginated endpoints
- **Why (was the change introduced):**
  - To provide inventory management capabilities for the ERP system, allowing tracking of items and their stock quantities across different parties/locations
- **How (the change was implemented):**
  - Created TypeORM entities with proper relationships (Item ↔ Inventory ↔ Party ↔ Company)
  - Implemented services with validation (unique HSN codes, unique item+party combinations)
  - Added filtering, sorting, and pagination support using existing decorator infrastructure
  - Soft delete implementation to preserve data integrity
Please check the type of change your PR introduces:
- [ ] Bug fix (non-breaking change)
- [x] New Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactor (no functional changes, no api changes)
- [ ] Build
- [ ] CI/ CD
- [ ] Documentation
- [ ] Security Patch
- [ ] Other (describe):
Please check the domain of the changes introduced in your PR:
- [ ] Frontend/ UI
- [x] Backend/ API
 Implementation Strategy
The implementation follows the existing NestJS module pattern:
- **Entities**: `Item`, `Inventory`, `Party`, `Company` with TypeORM decorators and relationships
- **DTOs**: Validation using `class-validator` with camelCase naming convention
- **Services**: Business logic with duplicate checks, soft delete, and referential integrity
- **Controllers**: REST endpoints with filter/sort/pagination decorators and response metadata
 How Has This Been Tested?
- Unit tests for `ItemsService` (12 tests) - all passing
- Unit tests for `ItemsController` (5 tests) - all passing
- Unit tests for `InventoryService` (17 tests) - all passing
- Unit tests for `InventoryController` (5 tests) - all passing
 API Changes (Backend/ API)
 Items Module (`/items`)
| Method | Endpoint | Description |
|--------|----------|-------------|
| POST | `/items` | Create a new item |
| GET | `/items` | List items with filtering, sorting, pagination |
| PATCH | `/items/:id` | Update an item |
| DELETE | `/items/:id` | Soft delete an item |
 Inventory Module (`/inventory`)
| Method | Endpoint | Description |
|--------|----------|-------------|
| POST | `/inventory` | Create inventory record |
| GET | `/inventory` | List inventory with filtering, sorting, pagination |
| PATCH | `/inventory/:id` | Update inventory record |
| DELETE | `/inventory/:id` | Soft delete inventory record |
 Developer Checklist
- [x] I have performed a self-review of my code
- [x] Tests have been added and all tests are passing
- [ ] Documentation has been updated (where necessary)
- [x] Code compiles/builds without errors or warnings
- [x] No sensitive information is exposed
- [ ] All dependent changes have been merged
 Reviewer Focus (optional)
- TODOs remain for permission checks (`InventoryItemChange`, `InventoryPartyChange`) when updating inventory records
- Item deletion is prevented if inventory records exist (referential integrity)

sharon504 committed:
feat(inventory): Adding items module

sharon504 committed:
feat(inventory): Added inventory module

sharon504 committed:
feat(inventory): Added paginated response type (temporary)

sharon504 committed:
refactor(inventory): Used util functions for filters

sharon504 committed:
fix(items): Updating items module, fixing variable naming issues, adding
company and party to inventory module

sharon504 committed:
Merge branch 'dev' into feature/inventory

sharon504 committed:
feat(inventory): Updating code to use response metadata setter

sharon504 committed:
fix(inventory): Removing all skipauthorization decorators

SreehariVinayan committed:
feat(vendor): scaffold vendor profile with company/party logic and mocks

- add /vendor page layout with company and party sections
- introduce CompanySection component with conditional rendering
- add scaffold party and company hooks and services
- define zod schemas for party and company
- add mock data and mock APIs for vendor flow development

alpha-og commented on apps/server/src/items/entities/item.entity.ts:
```diff
@@ -0,0 +1,50 @@
+  deletedAt?: Date;
+
+  @OneToMany(() => Inventory, (inventory) => inventory.item)
+  inventoryRecords!: Inventory[];
```
rename field to `inventory`

alpha-og commented on apps/server/src/items/items.controller.ts:
```diff
@@ -0,0 +1,93 @@
+      defaultField: "createdAt",
+      defaultDirection: "DESC",
+    })
+    order: FindOptionsOrder<Item> | undefined,
```
The `where` and `order` fields should be optional.

sharon504 committed:
refactor(inventory): Making formatting and naming changes

SreehariVinayan committed:
feat(vendor): complete company creation/edit flow and party update flow

- implement reusable CompanyForm for create and edit modes
- add company creation flow with conditional party create/update
- introduce PATCH party support and update hooks/services
- fix mock APIs and mock data
- mock state inspection via window.__MOCK_STATE__ for viewing changes in mock data (only for testing)
- refine vendor onboarding UX for no-party and no-company states

SreehariVinayan committed:
feat(vendor): support selecting existing companies and unlinking company from party

- add existing company selection via dropdown during company creation
- auto-fill and lock company fields when linking an existing company
- integrate GET /companies into company hook and services
- add party–company unlink action

parikz45 committed:
bid component

parikz45 committed:
bid-rpl

parikz45 committed:
updated issues

parikz45 committed:
updated issues

JVP555 committed:
feat(order): backend file structure created

JVP555 committed:
feat(party,company): moved to seperate folder/entities from inventory

JVP555 committed:
feat(party,company): moved to seperate folder/entities from inventory

JVP555 committed:
feat(party):dto for create and update of party created

JVP555 committed:
feat(party): added created by , created at , updated at and deleted at to party table

JVP555 committed:
feat(company): updated entity for update/delete details, created update and create dto

JVP555 committed:
feat(order): updated entity for update/delete details, created update and create dto

JVP555 committed:
feat(order-item): created order-item entity, created update and create dto

JVP555 committed:
fix imports

JVP555 committed:
restructured order-item into order

JVP555 committed:
restructured order-item into order

JVP555 committed:
restructured company into party and dto,entity of order-approval created

JVP555 committed:
feat(bid): entity and dtos of bid and bid approval created

SreehariVinayan committed:
feat(vendor-rpl): add admin registry page for companies and parties

- introduce /vendor-rpl admin page with companies and parties sections
- add CompaniesTable and PartiesTable
- implement view company modal with tabs (details / edit)
- support unlinking parties from companies and company deletion
- add company creation and edit flows
- fetch and display user details for party listings
- add vendor-rpl user hooks, services, schemas, and mock APIs

JVP555 committed:
feat(company,party): service for company and party created , user update in updatepartydto removed

adithyanrlal committed:
feat(order): implement controller and services for order management

JVP555 committed:
fix(company,party): service logic updated

JVP555 committed:
party.module.ts created

adithyanrlal committed:
fix(order-item): dtos updated

adithyanrlal committed:
feat(order-item): add basic controller and service for order item management

JVP555 committed:
feat(company,party):updated service to use find instead of findall and created company controller

JVP555 committed:
feat(party):created party controller

JVP555 committed:
fix(items):hsncode not unique

JVP555 committed:
feat(items): conflict based on hsncode removed and updated to use itemname instead , entity hsncode unique constraint removed and unique added for itemname

adithyanrlal committed:
feat(order-approval): add basic controller and service for order approval management

adithyanrlal committed:
feat(order): create module for order management

JVP555 committed:
fix(items): updated to use the correct import and methods for test of controller and service

adithyanrlal committed:
chore(order): organise controllers and services to seperate folders

adithyanrlal committed:
fix(order-approval): enhance update logic to include user authorization and add order status resolution

JVP555 reviewed (commented):
@alpha-og pls do check the current status of inventory and items based on the changes that was requested and if any more to resolve do inform

JVP555 committed:
Merge pull request #76 from RPL-ERP-CET/feature/order

Feature/order

JVP555 reviewed (changes requested):
not to merge till the issue 70 backend  is completed

JVP555 committed:
feat(order-item): approval logic incorporated to order-item

JVP555 committed:
updated entity for the one to many side in order with orderitems and updated order service

JVP555 committed:
Merge pull request #79 from RPL-ERP-CET/dev

Dev

JVP555 committed:
Merge pull request #80 from RPL-ERP-CET/feature/inventory

Feature/inventory

SreehariVinayan committed:
feat(inventory): implement paginated inventory table with mocks

- implement inventory service and hook
- add inventory feature schemas aligned with API response
- update table columns to consume nested inventory relations
- add mock inventory data

JVP555 committed:
Merge pull request #81 from RPL-ERP-CET/dev

Dev

JVP555 committed:
Merge pull request #82 from RPL-ERP-CET/feature/vendor

Feature/vendor

JVP555 committed:
frontend structure for purchase and sales created

JVP555 committed:
feat(order/schema):enum and entity type created

JVP555 committed:
chore(order):common feature to be filtered for sales and purchase

JVP555 committed:
removed unnecessary IsOptional

JVP555 committed:
feat(order/schemas):uses zod validation + infer to type

JVP555 committed:
chore(order): updated entity updatedto to use PartialType of CreateDto

JVP555 committed:
chore(order/schema):updated

JVP555 committed:
duplicate file

JVP555 committed:
barrel export for schemas

JVP555 committed:
party service updated as per frontend needs

JVP555 committed:
party service updated as per frontend needs

JVP555 committed:
width updated to use w-full

JVP555 committed:
updated to get parties that has no company linked

JVP555 committed:
updated to provide the service for party with null details

JVP555 committed:
updated with party remaining tasks

JVP555 committed:
Merge pull request #83 from RPL-ERP-CET/feature/frontend-inventory

feat(inventory): implement paginated inventory table with mocks

JVP555 committed:
deleted vendor mocks since now the party/company integration is completed

JVP555 committed:
columns for the inventory table

JVP555 committed:
items updated based on frontend requirements

JVP555 committed:
party updated based on frontend requirements

JVP555 committed:
created for the function to make visible data userfriendly

JVP555 committed:
updated to aauto create party in first visit

JVP555 committed:
update(seed): added new users for factory and estate party

JVP555 committed:
update(seed): added party role

JVP555 committed:
view/edit form for item

JVP555 committed:
form to link item and rpl

JVP555 committed:
column and table as per data table component

JVP555 committed:
page for inventory and item with crud

JVP555 committed:
updated hooks,schemas,services to work for inventory

JVP555 committed:
seeders for company party item and inventory added

JVP555 committed:
updated to show category and itemno

JVP555 committed:
updated todo for inventory

JVP555 committed:
Merge pull request #84 from RPL-ERP-CET/dev

Dev

JVP555 committed:
Merge pull request #85 from RPL-ERP-CET/dev

Dev

JVP555 committed:
reduntant

JVP555 committed:
Merge pull request #86 from RPL-ERP-CET/dev

Dev

JVP555 committed:
pushed for testing

alpha-og committed:
Merge branch 'dev' into feature/inventory

alpha-og committed:
fix: type errors

JVP555 committed:
Merge pull request #87 from RPL-ERP-CET/dev

Dev

JVP555 committed:
badge for ordertype

JVP555 committed:
badge for orderstatus

JVP555 committed:
barrel exports

JVP555 committed:
shifting to dataview

JVP555 committed:
refractoring

JVP555 committed:
seed for order

JVP555 committed:
updated to transform to date and time

JVP555 committed:
fix(order):updated service to return full response for the dataview

JVP555 committed:
feat(order): dataview component for order initialisaion initiated , unwanted files deleted

JVP555 committed:
order dataview implementation

JVP555 committed:
reverted back to original approval service

JVP555 committed:
feat(order): updated to use the order approval from order

JVP555 committed:
chore(order): updated to one page with tabs

JVP555 committed:
Merge pull request #89 from RPL-ERP-CET/dev

Dev

JVP555 committed:
badge components for party, amount and timestamp created

JVP555 committed:
barrel updated

JVP555 committed:
add order logic removed from dataview

JVP555 committed:
fix(order):updated fetch and cache logic

JVP555 committed:
Format of party name enhanced

JVP555 committed:
fix(order): inventory was incorrectly named as item , fixed

JVP555 committed:
feat(order-approval): badge for approval

JVP555 committed:
updated to supported the item -> inventory change

JVP555 committed:
updated to supported the item -> inventory change

JVP555 committed:
hook for getting itemnames as dropdown for orderitem enum

JVP555 committed:
added new utils for currency and normalize approval

JVP555 committed:
component to implement create lock logic

JVP555 committed:
feat(order): updated to use new create/edit logic of three sections(userfriendly approach)

JVP555 committed:
query key updated to work along with refresh and used the current badge components

JVP555 committed:
updated to supported the item -> inventory change

JVP555 committed:
updated to work with item->inventory change

JVP555 committed:
Comment component for custom styling of comments

JVP555 committed:
shows item name and passes inventory id

JVP555 committed:
updated to use the new dropdowns

JVP555 committed:
chore(order):new features

JVP555 committed:
chore(orderitem): create/view/edit completed

JVP555 committed:
chore(orderitem): delete completed / basic ui and functionality implemented

JVP555 committed:
chore(order): create/view/edit/delete completed

JVP555 committed:
chore(resubmit): resubmit logic implemented

JVP555 committed:
feat(item-options): dropdown for items which returns inventory id, extracted based on sourceID

JVP555 committed:
feat(user-options): list of users

JVP555 committed:
Duplicate(user):user service added for order

JVP555 committed:
fix(order): invalidation and refetching fixed

JVP555 committed:
fix(order-approval): comment box added for edit

JVP555 committed:
fix(order):approval and item uses datasignature as key to invalidate

JVP555 committed:
feat(status:order): options populated on the basis of type, util created for easy editing

JVP555 committed:
feat(order): party dropdown implemented

JVP555 committed:
Merge pull request #91 from RPL-ERP-CET/dev

Dev

JVP555 committed:
feat(order):ordermasterview creation initiated

JVP555 committed:
feat(order):ordermasterview creation initiated

JVP555 committed:
feat(order):orderitem masterview creation initiated

JVP555 committed:
feat(order): overview tab page adapted to master control

JVP555 committed:
feat(order): approval master initiated , refetch updated to work only with mutations

JVP555 committed:
refractor(order-approval): split into actions

JVP555 committed:
feat(order): order detail convered to master control

JVP555 committed:
feat(order): action route based on type

JVP555 committed:
feat(order): enhanced detail ui master control

JVP555 committed:
feat(order): enhanced detail ui master control

JVP555 committed:
fix(order): params corrected

JVP555 committed:
approver master control removed since its reduntant

JVP555 committed:
corrected page height

JVP555 committed:
Merge pull request #92 from RPL-ERP-CET/dev

Dev

JVP555 committed:
chore(order):updated order service and controller to use the new filter find

JVP555 committed:
master type updated to spread

JVP555 committed:
fix(approval): master removed

JVP555 committed:
corrected party fetch

JVP555 committed:
timeline corrected for purchase

JVP555 committed:
Purchase completed

JVP555 committed:
Errors corrected

JVP555 committed:
feat(approval) : dynamic calculation of status for order

JVP555 committed:
feat

JVP555 committed:
feat(inventory): inventory page updated to use latest dataview component

JVP555 committed:
opptional(inventory): schema order changed for better UI

JVP555 committed:
update(inventory): find method updated

JVP555 committed:
feat(Inventory): Dataview for inventory and items created

JVP555 committed:
feat(inventory): item dropdown added

JVP555 committed:
feat(items): crud completed

JVP555 committed:
fix: removed unused components

JVP555 committed:
fix: removed reduntant hook

JVP555 committed:
Deleted reduntant and shifted inventory dropdown inside inventory

JVP555 committed:
fix: cannot add approval without items correctly implemented

JVP555 committed:
feat(items): completed

JVP555 committed:
feat(inventory): Adding and subtracting creates production and consumption records respectively

JVP555 committed:
feat(inventory->order): as per stock adjustment order created

JVP555 committed:
feat(inventory->order): order+source+open then item added to that else new one, implemented on produce

JVP555 committed:
order bulk delete implemented with cascade for approval and items

JVP555 committed:
if status to completed then new order

JVP555 committed:
feat(Inventory): Completed , some validation and fixes remaining

JVP555 committed:
consumption , production greater than 0 validation

JVP555 committed:
removed current quantity from inventory create

JVP555 committed:
Merge pull request #102 from RPL-ERP-CET/dev

Dev

JVP555 committed:
changed from COMPLETED to CLOSED

JVP555 committed:
cleaned to work on vendor

JVP555 committed:
update(vendor): updated to use find method and bulk delete added

JVP555 committed:
feat(vendor): Dataview migration initialized

JVP555 committed:
updated barrel exports

JVP555 committed:
feat(vendor): updated get services to standard format and filter logic corrected

JVP555 committed:
update(vendor): dropdown import migrated

JVP555 committed:
feat(vendor): rpl side features link, unlink implemented

JVP555 committed:
Titles updated

JVP555 committed:
feat(Vendor): party crud completed

JVP555 committed:
feat(company): basic view updated and crud verified

JVP555 committed:
feat(vendor): added viewparties which filters parties from company table, fixed leak issue of filters

JVP555 committed:
updated app.module.ts

JVP555 committed:
chore: resolve merge conflicts and integrate vehicle/system-config features

JVP555 committed:
updated seed

JVP555 committed:
order factory updated with statusconfig map

JVP555 committed:
updated import of T_ApiSuccessResponse to reflect the shared types

JVP555 committed:
fix(inventory): issue of first item unknown fixed

JVP555 committed:
save

JVP555 committed:
inventory refracted to different pages

JVP555 committed:
Inventory added to sidebar

JVP555 committed:
filter and sort of inventory, items added

JVP555 committed:
order filter and sort added

JVP555 committed:
fixed dataview warning of order

JVP555 committed:
previous version is made hidden

JVP555 committed:
inventory verified

JVP555 committed:
updated

JVP555 committed:
Merge pull request #107 from RPL-ERP-CET/dev

Dev

JVP555 committed:
sidebar updated with order

JVP555 committed:
page updated

JVP555 committed:
feat(page): added pageheader of dataview container

JVP555 committed:
reduntant vendor-rpl removed

JVP555 committed:
feat(vendor): Added proper empty state

JVP555 committed:
updated masterview config

JVP555 committed:
ordermasterview UI updated

JVP555 committed:
masterview config tabs updated

JVP555 committed:
Merge pull request #108 from RPL-ERP-CET/dev

Dev

JVP555 committed:
added type indent

JVP555 committed:
indent added to sidebar

JVP555 committed:
added headoffice as a user

JVP555 committed:
feat(indent): Indent order draft creation completed

JVP555 committed:
Update Sidebar.tsx

JVP555 committed:
Update UserDropdown.tsx

JVP555 committed:
Delete apps/client-web/src/components/sidebar/sidebarConfig.ts
